### PR TITLE
feat: [CDS-68201]: retry failed pipeline option with last failed, selective stage and entire pipline added

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Moved duration and startTime upon hover

| Before | After |
| -- | -- |
| image | image |

- Retry Failed pipeline option combined with re-run pipeline ( Repeat icon removed with split button )
image
1. Re-run from last stage will pre-select last stage
2. Selected Stages(s) --> Retry failed pipelines
3. Entire pipeline --> Re-run pipeline

https://user-images.githubusercontent.com/100121280/236453801-01b7ca78-8bed-45aa-a165-7747cdcc07cb.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 1]
```diff
@@ -232,25 +285,46 @@ export function ExecutionHeader({ pipelineMetadata }: ExecutionHeaderProps): Rea
...
+                  />
+                  <SplitButtonOption
+                    data-test-id="retry-failed-pipeline-specific-stage"
+                    onClick={() => openRetryPipelineModal()}
+                    text={getString('pipeline.execution.actions.reRunSelectedStage')}
+                    disabled={isPipelineInvalid}
+                  />
+                </>
+              )}
+              <SplitButtonOption onClick={reRunPipeline} text={'Entire Pipeline'} />
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:320]: add text to strings

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 2]
```diff
@@ -146,6 +159,47 @@ export function ExecutionHeader({ pipelineMetadata }: ExecutionHeaderProps): Rea
...
+          color={Color.WHITE}
+        />
+        {pipelineExecutionSummary.startTs && (
+          <Text inline color={Color.WHITE} className={css.startTime}>
+            {`${getString('pipeline.startTime')}: ${formatDatetoLocale(pipelineExecutionSummary.startTs)}`}
+          </Text>
+        )}
+      </Layout.Vertical>
+    )
+  })
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:199]: `memo` is not useful if defined inside, move this out and pass props

**Review by rameshx — CHANGES_REQUESTED** [review]

**Comment by udayvunnam** [comment]
Is this only applicable on execution details page, how about execution ist page retry ?

**Comment by PratyushGar** [comment]
> Is this only applicable on execution details page, how about execution ist page retry ?

I'll check wth UX team whether we want to add similar option in Kebab menu there, as we don't have re-run button there

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 3]
```diff
@@ -59,6 +51,31 @@ export interface ExecutionHeaderProps {
   pipelineMetadata?: ResponsePMSPipelineSummaryResponse | null
 }
 
+interface TooltipContentProps {
+  pipelineExecutionSummary: PipelineExecutionSummary
+  getString: UseStringsReturn['getString']
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:56]: Nit: no need to pass getString as a prop, hook can be used inside

**Review by rameshx — APPROVED** [review]

**Review by SagarRajput7 — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Comment by PratyushGar** [comment]
image
Coverage is low for existing files

**Pull request merged**